src/web/mjs/connectors/VizShonenJump.mjs

```diff
@@ -0,0 +1,200 @@
+import Connector from '../engine/Connector.mjs';
+const EXIF = require('exif-js');
+
+export default class VizShonenJump extends Connector {
+
+    constructor() {
+        super();
+        super.id = 'vizshonenjump';
+        super.label = 'Viz - Shonen Jump';
+        this.tags = [ 'manga', 'english' ];
+        this.url = 'https://www.viz.com/shonenjump';
+        this.requestOptions.headers.set( 'x-referer', this.url );
+        this.config = {
+            /* Since the website requires us to fetch the image data a few seconds after the API call
+            *  we need to ensure we can download the data fast enough.
+            *  Also the website itself only loads a maximum of 4 pages at a time to ensure that.
+            */
+            concurrent: {
+                label: 'Concurrent downloads',
+                description: 'The maximum number of concurrent downloads.',
+                input: 'numeric',
+                min: 1,
+                max: 12,
+                value: 5
+            }
+        };
+        this.concurrent = 0;
+    }
+
+    async _getMangas() {
+        const request = new Request(new URL(this.url), this.requestOptions);
+        const data = await this.fetchDOM(request, 'div.o_sort_container div > a');
+
+        return data.map(manga => {
+            return {
+                id: manga.pathname,
+                title: manga.querySelector(':nth-child(2)').innerText.trim()
+            };
+        });
+    }
+
+    async _getChapters(manga) {
+        const request = new Request(new URL(manga.id, this.url), this.requestOptions);
+        let data = await this.fetchDOM(request, 'a.o_chapter-container');
+        data = data.filter(chapter => chapter.innerText.includes('FREE'));
+
+        return data.map(chapter => {
+            try {
+                return {
+                    id: chapter.href,
+                    title: chapter.querySelector('.disp-id').innerText.trim()
+                };
+            } catch(error) {
+                try {
+                    return {
+                        id: chapter.href,
+                        title: 'Ch. ' + chapter.href.match(/chapter-(\d+)\//)[1]
+                    };
+                } catch(error) {
+                    throw new Error('Unknown chapter format. Please report at https://github.com/manga-download/hakuneko/issues');
+                }
+            }
+        });
+    }
+
+    async _getPages(chapter) {
```

Comment:
I suggest to use `fetchRegex` instead of `fetchUI`, since it would consume less resources.
Something like this may work:
```javascript
// Some samples for development reference ...
// Mangas: https://www.viz.com/any/shonen-jump/section/45176/more
// Manga: https://www.viz.com/one-piece
// Chapters: https://www.viz.com/shonenjump/chapters/one-piece
// Chapter: https://www.viz.com/shonenjump/one-piece-chapter-1/chapter/5090
// Page: https://www.viz.com/manga/get_manga_url?device_id=3&manga_id=5090&page=1
let uri = new URL(manga.id, this.url); 
let request = new Request(...);
let data = await this.fetchRegex(request, /var\s+pages\s*=\s*(\d+)/g);
let pageCount = parseInt(data[0]);
let chapterID = 5090; // extracted from chapter url
return Array(17).fill().map((_, index) => {
    let page = new URL('https://www.viz.com/manga/get_manga_url');
    page.searchParams.set('device_id', 3); // 3 => probably for desktop with higher quality image link
    page.searchParams.set('manga_id', chapterID); // no idea why VIZ called it mangaID ...
    page.searchParams.set('page', index + 1);
    return this.createConnectorURI(page.href); // x-referer => this.url is sufficient, no need for chapter URL
});